### ISSUE TITLE
Status Code 400 with django

### ISSUE THREAD

**Issue by Dustlayer** [ecederstrand/exchangelib#691]
I'm trying to use exchangelib (version 2.1.1) in a django-environment (Django Version 2.2).
Python version 3.5.1.
The code I'm using works when executed manually. It also works when using the "manage.py shell" command.
It only doesn't work, when it is executed under "real web conditions".
I'm trying to accomplish that the user can fill a form, which in the backend creates a new meeting on the Exchange-server.
I even tried hardcoding my credentials, so it's not a problem of the form etc.
I also tried defining our company proxy via `class ProxyAdapter(requests.adapters.HTTPAdapter)`, but it even should work without it.

Testing `requests.get('https://mail.company.com/EWS/Exchange.asmx')` in the django code returns a `<Response [401]>`. So he should be able to access the server.

So the code in django:
```
from exchangelib import Credentials, Account, Configuration, DELEGATE
credentials = Credentials("DOMAIN\\username", "secretpassword")
config = Configuration(server="mail.company.com", credentials=credentials)
account = Account("name.name@company.com", config=config, autodiscover=False, access_type=DELEGATE)
```
This code works when I execute it manually.
But with django I keep getting these:
```
Unknown failure
Retry: 0
Waited: 10
Timeout: 120
Session: 26831
Thread: 140558522308576
Auth type: <requests_ntlm.requests_ntlm.HttpNtlmAuth object at 0x7fd6312c2be0>
URL: https://mail.company.com/EWS/Exchange.asmx
HTTP adapter: <mymodule.views.ProxyAdapter object at 0x7fd6312c2b38>
Allow redirects: False
Streaming: False
Response time: 0.06209137476980686
Status code: 400
Request headers: {'Content-Length': '0', 'Cookie': 'ClientId=KFUJLOEKFMY9WXSUBQ; expires=Fri, 11-Dec-2020 16:02:40 GMT; path=/; HttpOnly', 'Accept': '*/*', 'Accept-Encoding': 'gzip, deflate', 'User-Agent': 'exchangelib/2.1.1 (python-requests/2.22.0)', 'Content-Type': 'text/xml; charset=utf-8', 'Authorization': 'NTLM [...]/[...]', 'Connection': 'Keep-Alive'}
Response headers: {'Set-Cookie': 'exchangecookie=9c7db60d5de543a09966fbc0cb3c2bd2; expires=Sat, 12-Dec-2020 16:02:41 GMT; path=/; HttpOnly, X-BackEndCookie=S-1-5-21-1899719792-992224312-1087139693-182861=u56Lnp2ejJqBz8ucnZ7GnZ7Snc3LydLLnZrH0p7NyZvSmsfNyMmenJvOy8jIgYHNz83P0s/O0s7Oq87Jxc/NxcvP; expires=Sat, 11-Jan-2020 16:02:40 GMT; path=/EWS; secure; HttpOnly', 'Date': 'Thu, 12 Dec 2019 16:02:40 GMT', 'Content-Length': '0', 'X-Powered-By': 'ASP.NET', 'X-BEServer': 'SERVERNAME', 'X-DiagInfo': 'SERVERNAME', 'X-FEServer': 'OTHERSERVER', 'Server': 'Microsoft-IIS/8.5', 'request-id': 'aaa2a517-8151-4fc8-b61c-8d8c6dd3e3de', 'X-CalculatedBETarget': 'servername.domain.com', 'Persistent-Auth': 'true', 'Cache-Control': 'private', 'X-AspNet-Version': '4.0.30319'}
Request data: b''
Response data: b''
```

The output I'm seeing in the webserver logs:
```
No Windows timezone name found for timezone "CET". Fallback to UTC
EWS https://mail.company.com/EWS/Exchange.asmx, account None: Exception in _get_elements: Traceback (most recent call last):
  File "/path/to/virtualenv/lib/python3.5/site-packages/cached_property.py", line 69, in __get__
    return obj_dict[name]
KeyError: 'protocol'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/path/to/virtualenv/lib/python3.5/site-packages/exchangelib/services/common.py", line 71, in _get_elements
    response = self._get_response_xml(payload=payload)
  File "/path/to/virtualenv/lib/python3.5/site-packages/exchangelib/services/common.py", line 135, in _get_response_xml
    stream=self.streaming,
  File "/path/to/virtualenv/lib/python3.5/site-packages/exchangelib/util.py", line 738, in post_ratelimited
    _raise_response_errors(r, protocol, log_msg, log_vals)  # Always raises an exception
  File "/path/to/virtualenv/lib/python3.5/site-packages/exchangelib/util.py", line 812, in _raise_response_errors
    raise TransportError(str('Unknown failure\\n') + log_msg % log_vals)
exchangelib.errors.TransportError: Unknown failure
```

Have you got some ideas on this?

**Comment by ecederstrand** [comment]
For some reason, your request data is empty. That's not going to work. I have no idea why it's empty, so you'll have to start a debugger and hunt down the root cause.

**Comment by Dustlayer** [comment]
I tried debugging now but I can't really understand where the values are supposed to come from.
in services.common.py in `_get_response_xml` I tried analysing the parameters values.
I set a breakpoint and looked at the values, I'm guessing there interesting part is `data=wrap(content=payload, version=api_version, account=account)`.
So i wrote these to a file:
```
Content:
<Element {http://schemas.microsoft.com/exchange/services/2006/messages}GetFolder at 0x7fd6312becc8>
version:
Exchange2013_SP1
account:
dustin.buerstinghaus@obi.de
Wrap-Output:
b''
```

Have you got some advice where and how I can look where the values are gone missing?

**Comment by Dustlayer** [comment]
Short update:
The code works when I run it via django-webserver with `manage.py runserver`.
So it may be an issue with httpd (RHEL6).
I don't know how to find the root of the problem though.

**Comment by Dustlayer** [comment]
The issue is solved now. I'm afraid this issue is solved by "nvm fixed it".
We upgraded the server from RHEL6 to Oracle Linux 7. In the process we upgraded to Python 3.6 and from httpd 2.2 to httpd 2.4.
I can only tell that it works now.
Before upgrading I tried defining the version of the exchange server and the auth type explicitly. This brought me further. But as soon as a message was sent to the exchange server the request body/data was empty.
